.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,68 @@
+# GitHub Actions workflow to run tests on Linux.
+name: "Emscripten"
+
+on:
+  push: {}
+  pull_request: {}
+  schedule:
+    - cron: "0 0 * * *" # At 00:00 daily.
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: "Emscripten"
+    runs-on: "ubuntu-latest"
+    if: ${{ github.repository_owner == 'libressl' || github.event_name != 'schedule' }}
+    permissions:
+      contents: read
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+
+      - name: "Setup emsdk"
+        uses: mymindstorm/setup-emsdk@v14
+        with:
+          version: 3.1.60
```

Comment:
Using the latest emscripten package that is bundled with Debian 'trixie', this fails at build time like follows:

```
/home/bcook/projects/libressl/portable/crypto/cryptlib.c:373:21: error: variable has incomplete type 'struct syslog_data'
        struct syslog_data sdata = SYSLOG_DATA_INIT;
                           ^
/home/bcook/projects/libressl/portable/crypto/cryptlib.c:373:9: note: forward declaration of 'struct syslog_data'
        struct syslog_data sdata = SYSLOG_DATA_INIT;
               ^
/home/bcook/projects/libressl/portable/crypto/cryptlib.c:377:2: error: call to undeclared function 'vsyslog_r'; ISO C99 and later do not support implicit function declarations [-Werror,-Wimplicit-function-declaration]
        vsyslog_r(LOG_CONS|LOG_LOCAL2, &sdata, fmta, ap);
        ^
/home/bcook/projects/libressl/portable/crypto/cryptlib.c:377:2: note: did you mean 'vsyslog'?
/usr/share/emscripten/cache/sysroot/include/syslog.h:68:6: note: 'vsyslog' declared here
void vsyslog (int, const char *, va_list);
     ^
/home/bcook/projects/libressl/portable/crypto/cryptlib.c:386:16: error: call to undeclared function 'getprogname'; ISO C99 and later do not support implicit function declarations [-Werror,-Wimplicit-function-declaration]
            getuid(), getprogname(), file, line, assertion);
                      ^
/home/bcook/projects/libressl/portable/crypto/cryptlib.c:373:29: error: use of undeclared identifier 'SYSLOG_DATA_INIT'
        struct syslog_data sdata = SYSLOG_DATA_INIT;
                                   ^
4 errors generated.
```

Looks like it is an older version. Is it worth supporting, or should there be checks on version to make sure this fails earlier: 

```
$ dpkg -l |grep emscripten
ii  emscripten                                     3.1.6~dfsg-7                           all          LLVM-to-JavaScript Compiler
```

Hopefully #1055 isn't a generalization of the kinds of issues we might receive from the Emscripten community if this lands and doesn't work out of the box, or fail very explicitly, for everyone's environment.

Reply:
After conducting some research, I've determined that the oldest version this PR supports is 3.1.44 (released on 21, July, 2023). According to the Emscripten website, they recommend installing it via Git, so supporting older versions doesn't seem necessary as it's not the recommended method.

I can implement a version check using CMake. While it might not be the cleanest approach, it should work. This method involves invoking `emcc --version` and using regex to parse the version number. Here's a rough outline of the solution. Should I proceed with this approach? What do you think?

```cmake
# Execute the 'emcc --version' command and capture the output
execute_process(
    COMMAND emcc --version
    OUTPUT_VARIABLE EMSCRIPTEN_VERSION_OUTPUT
    OUTPUT_STRIP_TRAILING_WHITESPACE
)

# Parse the version string from the output
string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+" EMSCRIPTEN_VERSION ${EMSCRIPTEN_VERSION_OUTPUT})
```

Reply:
I'm inclined to not worry too much about it, and see if there are any reports in the wild. But we may want to add a check later if this becomes a common foot-gun of using ancient versions packaged with distributions.

Reply:
I feel the same way